api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -561,6 +556,13 @@ private void stopInitialCreditSpendInWorkspace(DbWorkspace workspace) {
         logger.error(
             "Failed to remove initial credits billing account from workspace {}", namespace, e);
       }
+    } else {
+      try {
+        leonardoApiClient.deleteAllResources(googleProject, false);
+        logger.info("Deleted apps and runtimes for workspace {}", namespace);
+      } catch (WorkbenchException e) {
+        logger.error("Failed to delete apps and runtimes for workspace {}", namespace, e);
```

Comment:
it actually would be good if we can call Leo when link billing failed (inside catch exception block above)